quickblox.min.js

```diff
@@ -42691,18 +42686,61 @@ RTCPeerConnection.prototype._startDialingTimer = function(extension, withOnNotAn
 /**
  * PRIVATE
  */
-function _getStats(peer, selector, successCallback, errorCallback) {
-    peer.getStats(selector, function (res) {
-        var items = [];
-        res.forEach(function (result) {
+function _getStats(peer, lastResults, successCallback, errorCallback) {
+    var statistic = {
+        'inbound_audio': null,
+        'inbound_video': null,
+        'outbound_audio': null,
+        'outbound_video': null,
+        'total_transport': null,
+    };
+
+    peer.getStats(null).then(function(results) {
+        results.forEach(function(result) {
             var item = {};
-            item.id = result.id;
-            item.type = result.type;
-            item.timestamp = result.timestamp;
-            items.push(item);
+            if (result.bytesSent && result.type === 'outbound-rtp') {
+                item.bitrate = _getBitratePerSecond(result, lastResults, true);
+                item.type = result.type;
+                item.mediaType = result.mediaType;
+                item.bytesSent = result.bytesSent;
+                item.packetsSent = result.packetsSent;
+                item.timestamp = result.timestamp;
+                statistic['outbound_' + result.mediaType] = item;
+            }
+            if (result.bytesReceived && result.type === 'inbound-rtp') {
+                item.bitrate = _getBitratePerSecond(result, lastResults, false);   
+                item.type = result.type;
+                item.mediaType = result.mediaType;
+                item.bytesReceived = result.bytesReceived;
+                item.packetsReceived = result.packetsReceived;
+                item.timestamp = result.timestamp;
+                statistic['inbound_' + result.mediaType] = item;
+            }
+            if (result.bytesSent && result.bytesReceived && result.type === 'transport') {
+                item.bitrate_out = _getBitratePerSecond(result, lastResults, true);
+                item.bitrate_in = _getBitratePerSecond(result, lastResults, false);    
+                item.type = result.type;
+                item.bytesSent = result.bytesSent;
+                item.bytesReceived = result.bytesReceived;
+                item.timestamp = result.timestamp;
+                statistic['total_' + result.type] = item;
+            }
         });
-        successCallback(items);
+        successCallback(statistic, results);
     }, errorCallback);
+
+    function _getBitratePerSecond(result, lastResults, isSent) {
+        var lastResult = lastResults && lastResults.get(result.id),
+            bitrate;
+        if (!lastResult) {
+            bitrate = 0;
+        } else if (isSent) {
+            bitrate = 8*(result.bytesSent-lastResult.bytesSent)/(result.timestamp-lastResult.timestamp);
```

Comment:
Could you use CONST for '8' with understanding name, please?

Reply:
Yes, OK